hand.py

```diff
@@ -0,0 +1,160 @@
+
+# Variable to track whether the screenshot has been captured
+screenshot_captured = False
+
+#Increase window dimensions to default size
+cv2.namedWindow("Hand gesture", cv2.WINDOW_NORMAL)
+cv2.resizeWindow("Hand gesture", window_width, window_height)
+
+while True:
+    _, img = cam.read()  # Read a frame from the webcam
+    img = cv2.flip(img, 1)
+    img_height, img_width, _ = img.shape
+    rgb_img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
+    
+    # Process the image to detect hands
+    output_hands = capture_hands.process(rgb_img)
+    all_hands = output_hands.multi_hand_landmarks
+    
+    if all_hands:
+        for i, hand in enumerate(all_hands):
+            # Generate random color for the hand
+            color = (random.randint(0, 255), random.randint(0, 255), random.randint(0, 255))
+            
+            # Draw landmarks on the hand with random color
+            drawing.draw_landmarks(img, hand, mediapipe.solutions.hands.HAND_CONNECTIONS, landmark_drawing_spec=drawing.DrawingSpec(color=color, thickness=2, circle_radius=4))
+            
+            # Get landmarks for one hand
+            one_hand_landmarks = hand.landmark
+            
+            for id, landmark in enumerate(one_hand_landmarks):
+                x = int(landmark.x * img_width)
+                y = int(landmark.y * img_height)
+
+                if id == 8:
+                    # Move mouse cursor to index finger position
+                    if use_two_fingers:
+                        mouse_x = int(screen_width / img_width * x)
+                        mouse_y = int(screen_height / img_height * y)
+                        pyautogui.moveTo(mouse_x, mouse_y, duration=0.4) #Reduces cursor speed
+                        x1 = x
+                        y1 = y
+                        cv2.circle(img, (x, y), 10, color, -1)
+                    else:
+                        # If using single finger (index), hide thumb (ID 4)
+                        x2 = -1
+                        y2 = -1
+                        x1 = x
+                        y1 = y
+                        mouse_x = int(screen_width / img_width * x)
+                        mouse_y = int(screen_height / img_height * y)
+                        pyautogui.moveTo(mouse_x, mouse_y)
+                        cv2.circle(img, (x, y), 10, color, -1)
+                    
+                if id == 4 and use_two_fingers:
+                    x2 = x
+                    y2 = y
+                    cv2.circle(img, (x, y), 10, color, -1)
+                
+
+        if use_two_fingers:
+            dist_y = y2 - y1
+        else:
+            # When using single finger, consider distance between the finger and the center of the screen
+            center_y = screen_height / 2
+            dist_y = y1 - center_y
+        
+        print(dist_y)
+        
+        # Determine gesture name based on the action being performed
+        gesture_name = ""
+        if dist_y > 0 and dist_y <= 35:
```

Comment:
Consider refactoring the code to separate gesture detection logic into functions for better modularity [medium]

- Generated by [LGTM](https://lgtm.pro)